src/main/java/ruiz/task/TaskList.java

```diff
@@ -0,0 +1,186 @@
+package ruiz.task;
+
+import java.io.IOException;
+import java.util.ArrayList;
+
+import ruiz.Parser;
+import ruiz.Ui;
+import ruiz.exception.BotException;
+
+/**
+ * Represents a list of tasks.
+ */
+public class TaskList {
+    private ArrayList<Task> taskList;
+    private Parser parser;
+    private Ui ui;
+
+    /**
+     * Constructor for the taskList class that is inititalised with a pre-saved taskList
+     *
+     * @param taskList List of tasks pre-saved in the hard disk
+     */
+    public TaskList(ArrayList<Task> taskList) {
+        this.taskList = taskList;
+        this.parser = new Parser();
+        this.ui = new Ui();
+    }
+
+    /**
+     * Constructor for the taskList class
+     */
+    public TaskList() {
+        this.taskList = new ArrayList<>();
+        this.parser = new Parser();
+        this.ui = new Ui();
+    }
+
+    public ArrayList<Task> getTaskList() {
+        return this.taskList;
+    }
+
+    /**
+     * Returns the size of the taskList.
+     *
+     * @return size of taskList.
+     */
+    public int getTaskListSize() {
+        return this.taskList.size();
+    }
+
+    /**
+     * This method marks the given task in the input.
+     *
+     * @param input The string that consists of the keyWord "mark" and task index being input by the user.
+     * @throws BotException if the input is not a valid one.
+     */
+    public String markTask(String input) throws BotException {
+        if (input.split(" ").length < 2) {
+            throw new BotException("OOPS!!! The index of a task cannot be empty.");
+        }
+        try {
+            int taskIndex = Integer.parseInt(input.split(" ")[1]) - 1;
+        } catch (NumberFormatException e) {
+            throw new BotException("OOPS!!! The index of a task has to be an integer.");
+        }
+        int taskIndex = Integer.parseInt(input.split(" ")[1]) - 1;
```

Comment:
Might want to abstract out the input parsing and error handling, to supply 'clean' indices as integers to the methods in TaskList. Note that this is not a coding standard violation.